src/embedded_flash/helper.rs

```diff
@@ -114,26 +107,23 @@ impl ModRange {
         new_length.map(|l| ModRange::new(self.start, l))
     }
 
-    /// Returns whether the given address is inside the range.
-    pub fn contains(&self, x: usize) -> bool {
-        // We want to check the 2 following inequalities:
-        // (1) `start <= x`
-        // (2) `x < start + length`
-        // However, the second one may overflow written as is. Using (1), we rewrite to:
-        // (3) `x - start <= length`
-        self.start <= x && x - self.start < self.length
-    }
-
-    /// Returns whether the given range is fully contained, i.e. their cut equals the parameter.
+    /// Returns whether the given range is fully contained.
+    ///
+    /// Mathematically, we calculate whether: `self ∩ range = range`.
     pub fn contains_range(&self, range: &ModRange) -> bool {
         if range.is_empty() {
             return true;
         }
-        let end = range.start + (range.length - 1);
-        self.contains(range.start) && self.contains(end)
+        range.is_empty()
+            || (self.start <= range.start
+                && range.length <= self.length
+                && range.start - self.start <= self.length - range.length)
     }
 
     /// Returns an iterator for all contained numbers that are divisible by the modulus.
+    ///
+    /// Be aware that `usize::MAX` is a special case for simplicity. It is never in the returned
+    /// iterator, even when divisible by `modulus` and contained in the range.
```

Comment:
I was curious to see what it would cost to get an implementation that supports MAX. It's not that bad.

```rust
let skip = (modulus - self.start % modulus) % modulus;
if self.start > MAX - skip || self.length < skip {
    (0..=0).take(0).step_by(1) // needs the same type as below
} else {
    (self.start + skip..=MAX).take(self.length - skip).step_by(modulus)
}
```

Reply:
Not bad, but I keep the simpler implementation I think!

Reply:
I actually have found the most readable and simple version:

```rust
// Iterate over all elements.
(self.start..=MAX)
    .take(self.length)
    // Skip the minimum number of elements to align.
    .skip((modulus - self.start % modulus) % modulus)
    // Only return aligned elements.
    .step_by(modulus)
```